manifest crashes with v8 serialization error in cluster mode

### 描述

运行 `pnpm run build:manifest` 时，如果构建器配置中包含有插件（如`githubRepoSyncPlugin`），构建失败

### 复现步骤

1. 使用有效的 S3 凭证配置 `.env`
2. 运行 `pnpm run build:manifest`

### 错误输出

```
20:36:10.324 Running "pnpm -w run build:manifest && pnpm --filter @afilmory/ssr build"
20:36:10.888 
20:36:10.889 > @afilmory/monorepo@0.1.0 build:manifest /vercel/path0
20:36:10.889 > pnpm --filter @afilmory/builder cli
20:36:10.889 
20:36:11.234 
20:36:11.236 > @afilmory/builder@0.2.2 cli /vercel/path0/packages/builder
20:36:11.236 > tsx src/cli.ts
20:36:11.236 
20:36:15.578 [info] [MAIN] 🚀 运行模式：增量更新
20:36:15.579 [info] [MAIN] ⚡ 最大并发数：4
20:36:15.580 [info] [MAIN] 🔧 处理模式：多进程集群
20:36:15.580 [info] [MAIN] 🏗️ 使用构建器：AfilmoryBuilder (适配器模式)
20:36:15.595 [info] [MAIN] Perl 已安装
20:36:15.601 [warn] [MAIN] ⚠️ 未配置远程仓库地址，跳过同步
20:36:15.602 [info] [MAIN] 🚀 开始从存储获取照片列表...
20:36:15.602 [info] [MAIN] 🔗 使用端点：...
20:36:15.602 [info] [MAIN] 🌐 自定义域名：...
20:36:15.602 [info] [MAIN] 🪣 存储桶：...
20:36:15.603 [info] [MAIN] 📂 前缀：gallery
20:36:15.604 [error] [FS] 🔍 未找到 manifest 文件/解析失败，创建新的 manifest 文件...
20:36:15.616 [info] [FS] 📁 Manifest 保存至： /vercel/path0/apps/web/src/data/photos-manifest.json
20:36:15.616 [info] [FS] 📷 包含 0 个相机，🔍 0 个镜头
20:36:15.617 [info] [MAIN] 现有 manifest 包含 0 张照片
20:36:15.617 [info] [MAIN] 使用存储提供商： s3
20:36:17.068 [info] [MAIN] 存储中找到 192 个文件
20:36:17.070 [info] [MAIN] 检测到 0 个 Live Photo
20:36:18.543 [info] [MAIN] 存储中找到 188 张照片
20:36:18.544 [info] [MAIN] 已按文件大小降序重排处理队列
20:36:18.545 [info] [MAIN] 存储中找到 188 张照片，实际需要处理 188 张
20:36:18.545 [info] [MAIN] 开始多进程处理任务，进程数：4，每进程并发：2
20:36:18.545 [info] [MAIN] 开始集群模式处理任务，进程数：4，总任务数：188
20:36:18.546 [info] [MAIN] 计算 worker 数量：总任务 188，每个 worker 并发 2，需要 94 个，实际启动 4 个
20:36:18.616 [start] [WORKER-2] Worker 2 进程启动 (PID: 437)
20:36:18.641 [start] [WORKER-3] Worker 3 进程启动 (PID: 438)
20:36:18.668 [start] [WORKER-1] Worker 1 进程启动 (PID: 436)
20:36:18.682 [start] [WORKER-4] Worker 4 进程启动 (PID: 439)
20:36:21.559 node:v8:456
20:36:21.560   ser.writeValue(value);
20:36:21.560       ^
20:36:21.560 
20:36:21.560 Error: async (context) => {
20:36:21.561         if (!enable) {
20:36:21.561           return;
20:36:21.561         }
20:36:21.561 
20:36:21.561         const { logger } = context;
20:36:21.562   ...<omitted>... } could not be cloned.
20:36:21.562     at serialize (node:v8:456:7)
20:36:21.562     at ClusterPool.handleWorkerReady (/vercel/path0/packages/builder/src/worker/cluster-pool.ts:270:34)
20:36:21.562     at Worker.<anonymous> (/vercel/path0/packages/builder/src/worker/cluster-pool.ts:203:20)
20:36:21.562     at Worker.emit (node:events:520:35)
20:36:21.563     at ChildProcess.<anonymous> (node:internal/cluster/worker:35:12)
20:36:21.563     at ChildProcess.emit (node:events:508:28)
20:36:21.563     at emit (node:internal/child_process:949:14)
20:36:21.563     at process.processTicksAndRejections (node:internal/process/task_queues:90:21)
20:36:21.564 
20:36:21.564 Node.js v24.13.0
20:36:21.688 /vercel/path0/packages/builder:
20:36:21.689  ERR_PNPM_RECURSIVE_RUN_FIRST_FAIL  @afilmory/builder@0.2.2 cli: `tsx src/cli.ts`
20:36:21.690 Exit status 1
20:36:21.733 Worker uncaught exception: Error: write EPIPE
20:36:21.734     at target._send (node:internal/child_process:877:20)
20:36:21.734     at target.send (node:internal/child_process:751:19)
20:36:21.734     at runAsWorker (/vercel/path0/packages/builder/src/runAsWorker.ts:327:13)
20:36:21.735     at main (/vercel/path0/packages/builder/src/cli.ts:20:11)
20:36:21.735     at <anonymous> (/vercel/path0/packages/builder/src/cli.ts:180:1)
20:36:21.735     at ModuleJob.run (node:internal/modules/esm/module_job:413:25)
20:36:21.735     at process.processTicksAndRejections (node:internal/process/task_queues:103:5)
20:36:21.735     at async onImport.tracePromise.__proto__ (node:internal/modules/esm/loader:660:26)
20:36:21.735     at async asyncRunEntryPointWithESMLoader (node:internal/modules/run_main:101:5) {
20:36:21.735   errno: -32,
20:36:21.736   code: 'EPIPE',
20:36:21.736   syscall: 'write'
20:36:21.736 }
20:36:21.759 Worker uncaught exception: Error: write EPIPE
20:36:21.760     at target._send (node:internal/child_process:877:20)
20:36:21.760     at target.send (node:internal/child_process:751:19)
20:36:21.760     at runAsWorker (/vercel/path0/packages/builder/src/runAsWorker.ts:327:13)
20:36:21.761     at main (/vercel/path0/packages/builder/src/cli.ts:20:11)
20:36:21.761     at <anonymous> (/vercel/path0/packages/builder/src/cli.ts:180:1)
20:36:21.761     at ModuleJob.run (node:internal/modules/esm/module_job:413:25)
20:36:21.761     at process.processTicksAndRejections (node:internal/process/task_queues:103:5)
20:36:21.761     at async onImport.tracePromise.__proto__ (node:internal/modules/esm/loader:660:26)
20:36:21.761     at async asyncRunEntryPointWithESMLoader (node:internal/modules/run_main:101:5) {
20:36:21.762   errno: -32,
20:36:21.762   code: 'EPIPE',
20:36:21.762   syscall: 'write'
20:36:21.762 }
20:36:21.770  ELIFECYCLE  Command failed with exit code 1.
20:36:21.797 Error: Command "pnpm -w run build:manifest && pnpm --filter @afilmory/ssr build" exited with 1
```

--- a/packages/builder/src/runAsWorker.ts
+++ b/packages/builder/src/runAsWorker.ts
@@ -1,10 +1,13 @@
+import { join } from 'node:path'
 import process from 'node:process'
+import { fileURLToPath } from 'node:url'
 import { deserialize } from 'node:v8'
 
 import type { PhotoManifestItem } from '@afilmory/typing'
 
 import type { BuilderOptions } from './builder/builder.js'
 import { AfilmoryBuilder } from './builder/builder.js'
+import { loadBuilderConfig } from './config/index.js'
 import type { PluginRunState } from './plugins/manager.js'
 import type { StorageObject } from './storage/interfaces'
 import type { BuilderConfig } from './types/config.js'
@@ -40,6 +43,17 @@ export async function runAsWorker() {
   let builder: AfilmoryBuilder
   let pluginRunState: PluginRunState
 
+  // 安全发送消息到主进程（防止 EPIPE 错误）
+  const safeSend = (message: unknown) => {
+    try {
+      if (process.send && process.connected) {
+        process.send(message)
+      }
+    } catch {
+      // 主进程已关闭 IPC 通道，静默忽略
+    }
+  }
+
   // 初始化函数，从主进程接收共享数据
   const initializeWorker = async (serializedData: WorkerInitMessage['sharedData']) => {
     if (isInitialized) return
@@ -52,7 +66,18 @@ export async function runAsWorker() {
     imageObjects = sharedData.imageObjects
     existingManifestMap = sharedData.existingManifestMap
     livePhotoMap = sharedData.livePhotoMap
-    builder = new AfilmoryBuilder(sharedData.builderConfig)
+
+    // 主进程序列化时移除了 plugins（函数无法序列化），
+    // 从配置文件重新加载以获取完整的插件配置
+    const fullConfig = await loadBuilderConfig({
+      cwd: join(fileURLToPath(import.meta.url), '../../../..'),
+    })
+    const builderConfig: BuilderConfig = {
+      ...sharedData.builderConfig,
+      plugins: fullConfig.plugins,
+    }
+
+    builder = new AfilmoryBuilder(builderConfig)
     await builder.ensurePluginsReady()
     pluginRunState = builder.createPluginRunState()
 
@@ -133,9 +158,7 @@ export async function runAsWorker() {
         result,
       }
 
-      if (process.send) {
-        process.send(response)
-      }
+      safeSend(response)
     } catch (error) {
       // 发送错误回主进程
       const response: TaskResult = {
@@ -144,9 +167,7 @@ export async function runAsWorker() {
         error: error instanceof Error ? error.message : String(error),
       }
 
-      if (process.send) {
-        process.send(response)
-      }
+      safeSend(response)
     }
   }
 
@@ -242,9 +263,7 @@ export async function runAsWorker() {
         results,
       }
 
-      if (process.send) {
-        process.send(response)
-      }
+      safeSend(response)
     } catch (error) {
       // 如果批量处理失败，为每个任务发送错误结果
       const results: TaskResult[] = message.tasks.map((task) => ({
@@ -258,9 +277,7 @@ export async function runAsWorker() {
         results,
       }
 
-      if (process.send) {
-        process.send(response)
-      }
+      safeSend(response)
     }
   }
 
@@ -275,9 +292,7 @@ export async function runAsWorker() {
 
       if (message.type === 'ping') {
         // 响应主进程的 ping，表示 worker 已准备好
-        if (process.send) {
-          process.send({ type: 'pong', workerId })
-        }
+        safeSend({ type: 'pong', workerId })
         return
       }
 
@@ -285,9 +300,7 @@ export async function runAsWorker() {
         // 处理初始化消息
         try {
           await initializeWorker(message.sharedData)
-          if (process.send) {
-            process.send({ type: 'init-complete', workerId })
-          }
+          safeSend({ type: 'init-complete', workerId })
         } catch (error) {
           console.error('Worker initialization failed', error)
           process.exit(1)
@@ -323,7 +336,5 @@ export async function runAsWorker() {
   })
 
   // 告知主进程 worker 已准备好
-  if (process.send) {
-    process.send({ type: 'ready', workerId })
-  }
+  safeSend({ type: 'ready', workerId })
 }

--- a/packages/builder/src/worker/cluster-pool.ts
+++ b/packages/builder/src/worker/cluster-pool.ts
@@ -267,11 +267,14 @@ export class ClusterPool<T> extends EventEmitter {
       // 首次准备就绪时发送初始化数据，但不立即标记为 ready
       if (this.sharedData) {
         // 使用 v8.serialize 序列化数据以保持类型完整性
+        // 注意：plugins 可能包含函数（如 hook 回调），v8.serialize 无法序列化函数
+        // 因此在序列化时移除 plugins，worker 进程会从配置文件重新加载
+        const { plugins: _plugins, ...serializableConfig } = this.sharedData.builderConfig
         const serializedBuffer = serialize({
           existingManifestMap: this.sharedData.existingManifestMap,
           livePhotoMap: this.sharedData.livePhotoMap,
           imageObjects: this.sharedData.imageObjects,
-          builderConfig: this.sharedData.builderConfig,
+          builderConfig: { ...serializableConfig, plugins: [] },
         })
 
         // 将 Buffer 转换为数组以通过 IPC 传输
@@ -515,7 +518,7 @@ export class ClusterPool<T> extends EventEmitter {
           const timeout = setTimeout(() => {
             worker.kill('SIGKILL')
             resolve()
-          }, 0)
+          }, 5000)
 
           worker.on('exit', () => {
             clearTimeout(timeout)

